service/awsconfig/v10/key/key.go

```diff
@@ -235,7 +238,16 @@ func MasterImageID(customObject v1alpha1.AWSConfig) string {
 }
 
 func MasterInstanceResourceName(customObject v1alpha1.AWSConfig) string {
-	return "MasterInstance"
+	s := strconv.FormatInt(time.Now().UnixNano(), 10)
+
+	h := sha1.New()
+	h.Write([]byte(s))
+	timeHash := fmt.Sprintf("%x", h.Sum(nil))[0:5]
+
+	upperTimeHash := strings.ToUpper(timeHash)
+	upperClusterID := strings.ToUpper(ClusterID(customObject))
+
+	return fmt.Sprintf("MasterInstance%s%s", upperClusterID, upperTimeHash)
```

Comment:
We generate the master resource name on each call.

Reply:
I think there should be a test for this. The output is not predictable but the format can be checked.

Reply:
Is checking the format important to you or that calls do not produce the same result?